core/render_management.ts

```diff
@@ -138,12 +140,23 @@ function updateConnectionLocations(block: BlockSvg, blockOrigin: Coordinate) {
  *
  * @param block The block to update the icon locations of.
  */
-function updateIconLocations(block: BlockSvg) {
+function updateIconLocations(block: BlockSvg, blockOrigin: Coordinate) {
   if (!block.getIcons) return;
   for (const icon of block.getIcons()) {
-    icon.computeIconLocation();
+    if (isIcon(icon)) {
+      icon.onLocationChange(blockOrigin);
+    } else {
+      // TODO (#7042): Remove old icon handling code.
+      icon.computeIconLocation();
+    }
   }
-  for (const child of block.getChildren(false)) {
-    updateIconLocations(child);
+  for (const conn of block.getConnections_(false)) {
```

Comment:
I don't understand this change -- doesn't `getChildren` already return the blocks connected via input and next connections? Using `getChildren` is more readable and already does some of the work like not finding previous connections and skipping connections with no blocks attached. I'm sure I'm missing something though.

Reply:
We could iterate the children, but we'd need to access their connections anyway to get the `relativeCoords` so we can do the fancy efficient math. So I think this is more readable given that constraint? but I can also check doing it the other way

Reply:
Isn't `target` === `child` (in the way the code was before)? 

So iiuc the code would be

```
for (const child of block.getChildren(false)) {
  updateIconLocations(target, Coordinate.sum(blockOrigin, child.relativeCoords))
}
```

Reply:
Oh whoops you're right I was reading this wrong. I thought `target` was `targetConnection` not `targetBlock`. Can fix =)